src/main/app/src/hexmap/PathFinder.js

```diff
@@ -1,87 +1,81 @@
+          return ndist < distance;
+        });
+      /*console.log(possibleSteps.length + " poss. steps (" +
                 possibleSteps.map(hex => hex.q + "," + hex.r).join("; ")
                 + ") in distance " + distance + " from " + source.q + "," + source.r
                 + " to " + target.q + "," + target.r);
              */
 
-            if (possibleSteps.length === 0) { //TODO wrong, wont go back
-                path.close()
-            } else {
-
-                //prolong path and create clones if there are more possibilties
-                let template = new TilePath();
-                template.tiles = path.tiles;
-                path.tiles.push(possibleSteps.shift());
-
-                for (let i = 0; i < possibleSteps.length; i++) {
-                    let clone = new TilePath();
-                    path.tiles.forEach(tile => clone.tiles.push(tile));
-                    clone.tiles.pop();
-                    //console.log("cloned path to add " + possibleSteps[i].q + "," + possibleSteps[i].r);
-                    //console.log("new clone:");
-                    //console.log(clone);
-                    clone.tiles.push(possibleSteps[i]);
-                    paths.push(clone);
-                }
-            }
-
-        });
-
-        //continue search if unclosed paths remain
-        if (paths.find(path => !path.closed) !== undefined) {
-            this.findPaths(paths, target)
-        } else {
-            this.sortAndFilterPaths(paths);
+      if (possibleSteps.length === 0) {
+        //TODO wrong, wont go back
+        path.close();
+      } else {
+        //prolong path and create clones if there are more possibilties
+        let template = new TilePath();
+        template.tiles = path.tiles;
+        path.tiles.push(possibleSteps.shift());
+
+        for (let i = 0; i < possibleSteps.length; i++) {
+          let clone = new TilePath();
+          path.tiles.forEach(tile => clone.tiles.push(tile));
+          clone.tiles.pop();
+          //console.log("cloned path to add " + possibleSteps[i].q + "," + possibleSteps[i].r);
+          //console.log("new clone:");
+          //console.log(clone);
+          clone.tiles.push(possibleSteps[i]);
+          paths.push(clone);
         }
-
-        //TODO pick one path, mark tiles as occupied to avoid path crossings
-
+      }
+    });
+
+    //continue search if unclosed paths remain
+    if (paths.find(path => !path.closed) !== undefined) {
+      this.findPaths(paths, target);
+    } else {
+      this.sortAndFilterPaths(paths);
     }
 
-    isOccupied(tile) {
-        return this.occupied.find(o => this.isSame(o, tile)) !== undefined;
-    }
+    //TODO pick one path, mark tiles as occupied to avoid path crossings
+  }
 
-    isSame(t1, t2) {
-        return t1.r === t2.r && t1.q === t2.q;
-    }
+  isOccupied(tile) {
+    return this.occupied.find(o => this.isSame(o, tile)) !== undefined;
+  }
 
+  isSame(t1, t2) {
```

Comment:
What's `t1``and `t2`? The parameters would benefit from a better name and/or a type.

Reply:
@bonndan